providers/openmetrics/client_interceptor.go

```diff
@@ -1,35 +1,21 @@
 package metrics
 
 import (
-	openmetrics "github.com/prometheus/client_golang/prometheus"
 	"google.golang.org/grpc"
 
 	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors"
 )
 
-// RegisterClientMetrics returns a custom ClientMetrics object registered
-// with the user's registry, and registers some common metrics associated
-// with every instance.
-func RegisterClientMetrics(registry openmetrics.Registerer) *ClientMetrics {
-	customClientMetrics := NewClientMetrics(registry)
-	customClientMetrics.MustRegister(customClientMetrics.clientStartedCounter)
-	customClientMetrics.MustRegister(customClientMetrics.clientHandledCounter)
-	customClientMetrics.MustRegister(customClientMetrics.clientStreamMsgReceived)
-	customClientMetrics.MustRegister(customClientMetrics.clientStreamMsgSent)
-
-	return customClientMetrics
-}
-
 // UnaryClientInterceptor is a gRPC client-side interceptor that provides Prometheus monitoring for Unary RPCs.
-func UnaryClientInterceptor(clientRegister openmetrics.Registerer) grpc.UnaryClientInterceptor {
+func UnaryClientInterceptor(clientMetrics *ClientMetrics) grpc.UnaryClientInterceptor {
```

Comment:
This was really weird. Interceptor shouldn't care about the registry, it only needs to increment the metrics. It shouldn't need to create it's own metrics and register them either.

Reply:
Agree, cc @yashrsharma44 for learning experience (: